src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<M extends Machine> {
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
If class MachineServiceImpl will be parameterized by Truck. Can we pass Bulldozer.class to the method getAll()? Is it valid behavior?
the same about other methods in this class